lib/config_cat/config_fetcher.ex

```diff
@@ -107,7 +154,34 @@ defmodule ConfigCat.CacheControlConfigFetcher do
 
   defp handle_response(%Response{status_code: code, body: config, headers: headers}, state)
        when code >= 200 and code < 300 do
-    with etag <- extract_etag(headers) do
+    with etag <- extract_etag(headers),
+         %{base_url: new_base_url, custom_endpoint: custom_endpoint} <-
+           Map.take(state, [:base_url, :custom_endpoint]),
+         p <- Map.get(config, Constants.preferences(), %{}),
+         base_url <- Map.get(p, Constants.preferences_base_url()),
+         redirect <- Map.get(p, Constants.redirect()) do
+      state =
+        cond do
```

Comment:
There is a missing part of this implementation. We prepared our SDKs to handle cases when we accidentally mess up our config.jsons. So to prevent redirect loops we only retry the fetch 3 times and if there is a redirect loop, we return with the last fetch's response and log a warning. 
https://github.com/configcat/ruby-sdk/blob/master/lib/configcat/configfetcher.rb#L120

Example:
cdn-global.configcat.com/.../config_v5.json:    { p: { r: 1, u: 'cdn-eu.configcat.com' }}
cdn-eu.configcat.com/.../config_v5.json:   { p: { r: 1, u: 'cdn-global.configcat.com' }}

Reply:
With the current implementation, I don't see this ever happening. We only follow the link if it's different then the link followed previously. And this is currently being validated with the current unit tests when we verify the number of calls even when the response is exactly the same as the one prior.

Reply:
Ah! Now I see what you mean... The config files itself are not configured properly so they will be redirecting to each other!